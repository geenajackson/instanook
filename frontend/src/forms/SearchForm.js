import React, { useState } from "react";
import { Button } from "reactstrap";

function SearchForm({ fetchData }) {
    const [search, setSearch] = useState("");

    const handleSubmit = e => {
        e.preventDefault();
        fetchData(search);
    };

    const handleChange = e => {
        setSearch(e.target.value);
    };

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="search"></label>
            <input
                id="search"
                name="search"
                type="text"
                placeholder="Search"
                onChange={handleChange}>
            </input>
            <Button size="sm" type="submit">Search!</Button>
        </form>
    )
}

export default SearchForm;