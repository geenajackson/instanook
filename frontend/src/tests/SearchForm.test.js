import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "../forms/SearchForm";

it("renders without crashing", function () {
    render(<SearchForm />);
});

it("matches snapshot with no jobs", function () {
    const { asFragment } = render(<SearchForm />);
    expect(asFragment()).toMatchSnapshot();
});