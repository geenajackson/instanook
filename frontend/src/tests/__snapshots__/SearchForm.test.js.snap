// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with no jobs 1`] = `
<DocumentFragment>
  <form>
    <label
      for="search"
    />
    <input
      id="search"
      name="search"
      placeholder="Search"
      type="text"
    />
    <button
      class="btn btn-secondary btn-sm"
      type="submit"
    >
      Search!
    </button>
  </form>
</DocumentFragment>
`;
